ios/Gekidou/Sources/Gekidou/FileCache.swift

```
@@ -0,0 +1,64 @@
+//
+//  FileCache.swift
+//  Gekidou
+//
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+import Foundation
+import UIKit
+
+public class FileCache: NSObject {
+  private var cacheURL: URL?
+  @objc public static let `default` = FileCache()
+  
+  override private init() {
+    super.init()
+    let filemgr = FileManager.default
+    let appGroupId = Bundle.main.infoDictionary!["AppGroupIdentifier"] as! String
+    let containerUrl = filemgr.containerURL(forSecurityApplicationGroupIdentifier: appGroupId)
+    if let url = containerUrl,
+    let cacheURL = url.appendingPathComponent("Library", isDirectory: true) as URL? {
+      self.cacheURL = cacheURL.appendingPathComponent("Caches", isDirectory:  true)
+      self.createDirectoryIfNeeded(directory: self.cacheURL)
+    }
+  }
+  
+  private func createDirectoryIfNeeded(directory: URL?) {
+    var isDirectory = ObjCBool(false)
+       
+    if let cachePath = directory?.path {
+      let exists = FileManager.default.fileExists(atPath: cachePath, isDirectory: &isDirectory)
+      
+      if !exists && !isDirectory.boolValue {
+        try? FileManager.default.createDirectory(atPath: cachePath, withIntermediateDirectories: true, attributes: nil)
```

it fails silently if it cannot create the cache directory